pkg/ebpf/tracer.go

```diff
@@ -434,3 +472,181 @@ func (m *FlowFetcher) LookupAndDeleteMap() map[BpfFlowId]*BpfFlowMetrics {
 	}
 	return flow
 }
+
+// It provides access to packets from  the kernel space (via PerfCPU hashmap)
+type PacketFetcher struct {
+	objects        *BpfObjects
+	qdiscs         map[ifaces.Interface]*netlink.GenericQdisc
+	egressFilters  map[ifaces.Interface]*netlink.BpfFilter
+	ingressFilters map[ifaces.Interface]*netlink.BpfFilter
+	perfReader     *perf.Reader
+	cacheMaxSize   int
+	enableIngress  bool
+	enableEgress   bool
+}
+
+func NewPacketFetcher(
+	cacheMaxSize int,
+	pcaFilters string,
+	ingress, egress bool,
+) (*PacketFetcher, error) {
+	if err := rlimit.RemoveMemlock(); err != nil {
+		log.WithError(err).
+			Warn("can't remove mem lock. The agent could not be able to start eBPF programs")
+	}
+
+	objects := BpfObjects{}
+	spec, err := LoadBpf()
+	if err != nil {
+		return nil, err
+	}
+
+	pcaPort := 0
+	pcaProto := 0
+	filters := strings.Split(pcaFilters, ",")
+	if filters[0] == "tcp" {
+		pcaProto = syscall.IPPROTO_TCP
+	} else {
+		pcaProto = syscall.IPPROTO_UDP
+	}
+	pcaPort, _ = strconv.Atoi(filters[1])
+
+	if err := spec.RewriteConstants(map[string]interface{}{
+		constPcaPort:  uint32(pcaPort),
+		constPcaProto: uint8(pcaProto),
+	}); err != nil {
+		return nil, fmt.Errorf("rewriting BPF constants definition: %w", err)
+	}
+	plog.Debugf("PCA Filters: %d, %d Incoming: %s", pcaProto, pcaPort, pcaFilters)
```

Comment:
Maybe raise as Info log, as it's a one-time log that helps to retrieve the configuration applied